pkg/proxy/userspace/proxier.go

```diff
@@ -35,8 +35,9 @@ import (
 )
 
 type portal struct {
-	ip   net.IP
-	port int
+	ip         net.IP
+	port       int
+	isExternal bool
```

Comment:
don't stick this in the portal struct - just pass it as an extra param to openOnePortal